src/sqlparser/src/ast/value.rs

```diff
@@ -60,7 +60,9 @@ pub enum Value {
     /// `NULL` value
     Null,
     /// name of the reference to secret
-    Ref(SecretRef),
+    SecretRef(SecretRefValue),
+    /// name of the reference to connection
+    ConnectionRef(ConnectionRefValue),
```

Comment:
Don't do this. Here the `struct Value` means a general value in SQL and you can observe this from the rest variants. `SecretRef` and `ConnectionRef` are definitely shouldn't be here.

cc. @yuhao-su Please change the secret's implementation as well.

Reply:
Maybe refactor it into a RefValue, meaning refer to other items in catalog?

Reply:
Hmm I mean it should not touch the `struct Value` at all. Perhaps we need to modify `SqlOption` by allows `value` to be one of `Value`, secret or connection.

https://github.com/risingwavelabs/risingwave/blob/b6051d84c7e7d5dc4bf3994d1ea95ca93166723f/src/sqlparser/src/ast/statement.rs#L875

https://github.com/risingwavelabs/risingwave/blob/45a9b8db58f57fbe367705f6d93f120fc33ddbf7/src/sqlparser/src/ast/mod.rs#L2735-L2738

Reply:
will resolve via https://github.com/risingwavelabs/risingwave/pull/19559